### PR TITLE
feat: for compatibility, code also have language

### PR DESCRIPTION
**Pull request by zaratan**
Some PrismJS themes expect `<code>` to also carry the language.

### CODE REVIEW

**Comment by hswolff** [comment]
Would love to see this merged! When porting from Gatsby to Next.js I noticed this as a difference between how the two rendered.

**Comment by zaratan** [comment]
@hswolff that's exactly why I created that PR. I'm using my fork now and everything is working perfectly now.

**Comment by sergioramos** [comment]
Thank you so much for this @zaratan 👍

I just have a couple of questions, and will merge if you don't mind answering.

File: index.js [hunk 1]
```diff
@@ -162,7 +162,10 @@ const parseLang = (str) => {
     .replace(/\}$/, '');
 
   const attrs = selectors.length ? selectorToAttrs(selectors) : {};
-  const className = classNames(lang ? `language-${lang}` : '', attrs.class);
+  const className = classNames(
+    lang ? `language-${lang}` : 'language-',
```
Comment (sergioramos) [inline index.js:166]: any reason why not an empty string?

File: index.js [hunk 2]
```diff
@@ -202,7 +205,7 @@ module.exports = (options = {}) => (tree) => {
         );
       }, '');
 
-    const code = h('code', {}, u('raw', raw));
+    const code = h('code', { className: `language-${lang}` }, u('raw', raw));
```
Comment (sergioramos) [inline index.js:208]: what if lang is undefined? wouldn't that add `language-undefined`?

Comment (zaratan) [reply index.js:208]: You are right. Let me fix this.

File: index.js (hunk 1 above)
Comment (zaratan) [reply index.js:166]: I agree that it looks like a bad thing but sadly prism themes often use this kind of checks:
```css
code[class*="language-"], pre[class*="language-"] {
```
See: https://github.com/xiaozi/solarized-prism-theme/blob/c6d880bfeae792c1582c3f07071407ac9d17794c/solarized-dark.css#L6

If you have no language specified (which will always be the case with inline code blocks for example) those theme might not work correctly.

**Comment by zaratan** [comment]
That said. Maybe this should be put behind a `compatible_mode` option flag. What do you think @sergioramos ?

**Comment by zaratan** [comment]
@sergioramos `language-unknown` seems better than `language-` and those theme would still work. IMO we don't need a `compatibility` flag on top of this then.

**Comment by sergioramos** [comment]
thank you very much @zaratan 👍

**Comment by zaratan** [comment]
Thank you for this project. It works very well :)

**Pull request merged**